enhancements/declarative-index-config.md

```diff
@@ -481,40 +616,119 @@ The representation of the content of an index using json/yaml provides a unique
+| `olm.gvk.provided`     | `{ group, version, kind string }`      | `{"type":"olm.gvk.provided", "value": {"group": "etcd.database.coreos.com", "version": "v1beta2", "kind": "EtcdBackup"}}` |
+| `olm.gvk.required`     | `{ group, version, kind string }`      | `{"type":"olm.gvk.required", "value": {"group": "test.coreos.com", "version": "v1", "kind": "Testapi"}}`                  |
+| `olm.skips`            | `string`                               | `{"type":"olm.skips", "value": "etcdoperator.v0.9.0"}`<br>`{"type":"olm.skips", "value": "etcdoperator.v0.9.2"}`          |
+| `olm.skipRange`        | `string`                               | `{"type":"olm.skipRange", "value": "<0.9.4"}`                                                                             |
+| `olm.channel`          | `{ name, replaces string }`            | `{"type":"olm.channel", "value":{"name":"singlenamespace-alpha", "replaces":"etcdoperator.v0.9.2"}}`<br>`{"type":"olm.channel", "value":{"name":"clusterwide-alpha"}}`         |
+
+To support both old and new clients, `opm` will ensure that there are matching `olm.gvk`/`olm.gvk.provided` and `olm.package`/`olm.package.provided` properties present in bundle properties when migrating from the sqlite database representation and also during build time and validation. When all supported clients have been transitioned to understand the new property names, `olm.gvk` and `olm.package` will be retired.
+Since the `olm.bundle` schema also has a top-level field for `package`, `opm` will verify that it aligns with the `olm.package`/`olm.package.provided` properties.
+
+#### Filesystem structure
+
+Declarative configs are expected to exist in a directory structure. When reading a declarative config index, `opm` will
+traverse the file tree and load valid blobs from each file it finds. When writing a declarative config, package-specific
+blobs are concatenated into a single `<packageName>.json` file, and non-package-specific blobs are concatenated into
+`__global.json`.
+
+The output directory structure will be:
+```bash
+$ tree <index_name>
+<index_name>
+├── amqstreams.json
+├── etcd.json
+├── __global.json
+└── servicemesh.json
+```
+
+When stored in a tarball, the output structure will be:
+```bash
+$ tar tvf <index_name>.db
+drwxr-xr-x nobody/nobody     0 2021-03-17 17:18 index/
+-rw-r--r-- nobody/nobody     0 2021-03-17 15:44 index/servicemesh.json
+-rw-r--r-- nobody/nobody     0 2021-03-17 17:18 index/__global.json
+-rw-r--r-- nobody/nobody     0 2021-03-17 15:39 index/amqstreams.json
+-rw-r--r-- nobody/nobody     0 2021-03-17 15:38 index/etcd.json
+```
+
+When migrating an sqlite file to a tarball, `opm` will create the tarball with the declarative directory structure
+(described above) with the root directory always called `index`.
+
+For example, if `my-index.db` is an sqlite file `opm registry add --database /path/to/my-index.db --bundle my-bundle:v0.1.0`
```

Comment:
Reading this it's not clear to me what value we're getting in doing this? What's the benefit of the tarball over just replacing it with a directory of configs named `my-index.db`?